evennia/web/website/forms.py

```diff
@@ -0,0 +1,142 @@
+            min_points (int): The least amount of points each bucket should have.
+            max_points (int): The most points each bucket should have.
+            
+        Returns:
+            buckets (list): List of random point assignments.
+        
+        """
+        buckets = [0 for x in range(num_buckets)]
+        indices = [i for (i, value) in enumerate(buckets)]
+    
+        # Do this while we have eligible buckets, points to assign and we haven't
+        # maxed out all the buckets.
+        while indices and points and sum(buckets) <= (max_points * num_buckets):
+            # Pick a random bucket index
+            index = choice(indices)
+           
+            # Add to bucket
+            buckets[index] = buckets[index] + 1
+            points = points - 1
+            
+            # Get the indices of eligible buckets
+            indices = [i for (i, value) in enumerate(buckets) if (value < min_points) or (value < max_points)]
+            
+        return buckets
+    
+class ExtendedCharacterCreationForm(forms.Form):
+    
+    GENDERS = (
+        ('male', 'Male'),
+        ('female', 'Female'),
+        ('androgynous', 'Androgynous'),
+        ('special', 'Special')
+    )
+    
+    RACES = (
+        ('human', 'Human'),
+        ('elf', 'Elf'),
+        ('orc', 'Orc'),
+    )
+    
+    CLASSES = (
+        ('civilian', 'Civilian'),
+        ('warrior', 'Warrior'),
+        ('thief', 'Thief'),
+        ('cleric', 'Cleric')
+    )
+    
+    PERKS = (
+        ('strong', 'Extra strength'),
+        ('nimble', 'Quick on their toes'),
+        ('diplomatic', 'Fast talker')
+    )
+    
+    name = forms.CharField(help_text="The name of your intended character.")
+    age = forms.IntegerField(min_value=3, max_value=99, help_text="How old your character should be once spawned.")
+    gender = forms.ChoiceField(choices=GENDERS, help_text="Which end of the multidimensional spectrum does your character most closely align with, in terms of gender?")
+    race = forms.ChoiceField(choices=RACES, help_text="What race does your character belong to?")
+    job = forms.ChoiceField(choices=CLASSES, help_text="What profession or role does your character fulfill or is otherwise destined to?")
+    
+    perks = forms.MultipleChoiceField(choices=PERKS, help_text="What extraordinary abilities does your character possess?")
+    description = forms.CharField(widget=forms.Textarea(attrs={'rows': 3}), max_length=2048, min_length=160, required=False)
+    
+    strength = forms.IntegerField(min_value=1, max_value=10)
+    perception = forms.IntegerField(min_value=1, max_value=10)
+    intelligence = forms.IntegerField(min_value=1, max_value=10)
+    dexterity = forms.IntegerField(min_value=1, max_value=10)
+    charisma = forms.IntegerField(min_value=1, max_value=10)
+    vitality = forms.IntegerField(min_value=1, max_value=10)
+    magic = forms.IntegerField(min_value=1, max_value=10)
+    
+    def __init__(self, *args, **kwargs):
```

Comment:
I think this whole extended character creation form rather belongs either in a contrib or explained in a tutorial. It might also be good to note that users of telnet will be left out if this is the main way to create a character - one would need to make a separate character creation mechanism for those not using the web interface.

Reply:
Agreed; this is a bit of a mess currently with a lot of nuances to work through. Ultimately we need a common interface for creating and validating characters and directing telnet or webserver requests for new characters through it so there are no inconsistencies.